src/main/java/org/jabref/model/entry/AuthorList.java

```diff
@@ -647,10 +566,14 @@ public String getAsFirstLastNamesLatexFree(boolean abbreviate, boolean oxfordCom
     /**
      * Compare this object with the given one.
      * <p>
-     * Will return true iff the other object is an Author and all fields are identical on a string comparison.
+     * @return `true` iff the other object is an AuthorList, all Authors are in the same order, and all Authors field
```

Comment:
```suggestion
     * @return `true` iff the other object is an AuthorList, all contained authors are in the same order (and these authors are `String.equal`)
```

Reply:
I'd suggest changing the parenthesis to "(and these authors' fields are `Objects.equals`)?
I'll update the JavaDoc in `Author.java` as well.

Reply:
Good idea!

I already commited the change to `checkstyle.xml`.

After that udpate, this PR is good to go.